python/semantic_kernel/utils/settings.py

```diff
@@ -1,8 +1,55 @@
 # Copyright (c) Microsoft. All rights reserved.
 
-from typing import Dict, Optional, Tuple, Union
+from typing import Dict, List, Literal, Optional, Tuple, Union
 
 from dotenv import dotenv_values
+from pydantic import field_validator
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+
+class AzureChatCompletionSettings(BaseSettings):
+    model_config = SettingsConfigDict(
+        env_file=".env", env_prefix="AZURE_OPENAI_", extra='ignore')
```

Comment:
I really like the idea of instantiating the settings directly in the object itself -- it sounds like a cleaner way to do it.  

Based on what I'm seeing in the [docs about dotenv support](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#dotenv-env-support) for pydantic settings, it seems that environment variables will always take priority over dotenv settings.  Here is the relevant line in the docs since I can't link to it directly, but you should be able to find it by scrolling down a bit from the link.  

> Even when using a dotenv file, pydantic will still read environment variables as well as the dotenv file, environment variables will always take priority over values loaded from a dotenv file.

I think for now I can remove this from this PR, so I can create a separate PR to make it more clean and take the feedback into account.  I mainly just wanted to get some feedback on the idea, which is why I included it here to start with.  Thanks for your thoughts about it!

Reply:
> @@ -208,10 +209,12 @@ def __init__( if base_url and isinstance(base_url, str): base_url = HttpsUrl(base_url) if use_extensions and endpoint and deployment_name: - base_url = HttpsUrl(f"{str(endpoint).rstrip('/')}/openai/deployments/{deployment_name}/extensions") + base_url = HttpsUrl(